### PR TITLE
Use a custom condition to break out of an R analysis

### PR DESCRIPTION
**Pull request by TimKDJ**
Fixes https://github.com/jasp-stats/INTERNAL-jasp/issues/462

jaspResults will be able to break out of try statements, e.g.,
```
startProgressbar(1000)
try({
  for (i in 1:1000) {
    Sys.sleep(0.1)
    progressbarTick()
  }
})
<this point will not be reached>
```

### CODE REVIEW

File: JASP-R-Interface/jaspResults/src/jaspResults.cpp [hunk 1]
```diff
@@ -216,8 +214,8 @@ void jaspResults::checkForAnalysisChanged()
 	{
 		jaspPrint("Polling for analysis changes found a change, analysis should restart!");
 		setStatus("changed");
-		static Rcpp::Function stop("stop");
-		stop(_analysisChangedErrorMessage);
+		static Rcpp::Function signalAnalysisAbort = jaspResults::isInsideJASP() ? Rcpp::Function("signalAnalysisAbort") : Rcpp::Environment::namespace_env("jaspResults")["signalAnalysisAbort"];
```
Comment (JorisGoosen) [inline JASP-R-Interface/jaspResults/src/jaspResults.cpp:217]: Funky!

**Comment by JorisGoosen** [comment]
Looks good!

You will need to increment the JASP R Interface version number though...

**Comment by TimKDJ** [comment]
> Looks good!
> 
> You will need to increment the JASP R Interface version number though...

I guess I'm going to need a post-it in the middle of my screen reminding me of this...

**Comment by JorisGoosen** [comment]
Yeah that is annoying, I'll check and see if we can get rid of the whole JASP-R-Interface being split off from the rest. Then you can just use your screen for it's display properties :p

**Pull request merged**